eulerpi/core/model.py

```diff
@@ -96,6 +96,21 @@ def forward_and_jacobian(
 
         return self.forward(param), self.jacobian(param)
 
+    def param_is_within_domain(self, param: np.ndarray) -> bool:
+        """Checks whether a parameter is within the parameter domain of the model.
+        Overwrite this function if your model has a more complex parameter domain than a box.
```

Comment:
... than a box. The param_limits are checked automatically.

You can link to the param_limits variable also.